hubblestack/hangtime.py

```diff
@@ -1,23 +1,74 @@
+# -*- coding: utf-8 -*-
+'''
+Module for handling timeouts in code that may not be designed for it.
+'''
 
 import signal
 import time
 
 import logging
 log = logging.getLogger('hangtime')
 
-def _alive(x):
-    try:
-        return x.id is not None
-    except ReferenceError:
-        return False
-
 class HangTime(Exception):
     # NOTE: this will break completely in multithreading
     # it should work just fine in multiprocessing
 
     prev = list()
 
     def __init__(self, msg="hang timeout detected", timeout=300, id=0, repeats=False, decay=1.0):
+        ''' HangTime wraps code via with block.
+
+        ... code-block:: python
```

Comment:
I think it's one leading dot. Though I'm tempted to go with markdown instead of RST....though we should probably match salt, I just don't love RST.